pkg/capacity/capacity.go

```diff
@@ -299,7 +299,7 @@ func (c *Controller) onTopologyChanges(added []*topology.Segment, removed []*top
 			continue
 		}
 		if !c.immediateBinding && sc.VolumeBindingMode != nil && *sc.VolumeBindingMode == storagev1.VolumeBindingImmediate {
-			return
+			continue
```

Comment:
Can we add a unit test for this?

Reply:
We can, but it's not perfect: we need a specific ordering of storage classes (first the one with immediate binding, then the one with late binding), and there's no guarantee that the informer will list them in any particular order. But in practice, the informer seems to use "first in, first out", so I was able to trigger the problem.

But worse, `onTopologyChanges` wasn't tested at all. Fixed...